lib/data_hygiene/data_sync.rb

```diff
@@ -0,0 +1,152 @@
+module DataHygiene
+  class DataSync
+    PRODUCTION_ACCOUNT_CODE = "UKGOVUK"
+    THREAD_COUNT = 40
+
+    attr_reader :logger
+
+    def initialize(logger = Logger.new(STDOUT), delete_wait = 30)
+      @logger ||= logger
+      @delete_wait = delete_wait
+    end
+
+    def run
+      validate!
+
+      update_delivery_id!
+
+      logger.info "Retrieving data from servers.."
+
+      # determine which topics/subscriber_lists already exist and which have been
+      # modified/created since the last data sync
+      matching = topics & subscriber_lists
+      to_be_deleted = topics - subscriber_lists
+      to_be_created = subscriber_lists - topics
+
+      # filter out any duplicate ID with at least one match from the list to be delete/created
+      # - this avoids errors during creation and avoid unnecessary deletions and creations
```

Comment:
I think this is to handle rows with duplicate codes but different titles, because one of those may exist in GovDelivery already so we don't want to try to create/delete another topic with the same id - is that right? Could you expand this comment? It took me a while to work out why we'd need to do this.